### ISSUE TITLE
[FX] Write a GPU inference optimization pass

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#48209]
We should have a pass `prepare_for_gpu_inference(m : nn.Module) -> nn.Module` (or something similar) that will do things that will make a given model run faster on GPU in the inference scenario. The stages of this pipeline should include:

* FP16 conversion of the model
* Convert the model to NHWC layout
* Fuse patterns like Conv-BN and Conv-BN-ReLU

And we should have documentation for this pass, particularly pointing out that the datatype change may change accuracy, so users should re-eval their model after they run it.

With this pass, we estimate there should be a ~2x latency improvement on small batches on V100 and A100 GPUs
